FrameworkSystem/Client/ProxyManagerClient.py

```diff
@@ -147,43 +153,34 @@ def uploadProxy(self, proxy=False, diracGroup=False, chainToConnect=False, restr
       if not result['OK']:
         return S_ERROR("Can't load %s: %s " % (proxyLocation, result['Message']))
 
-    if not chainToConnect:
-      chainToConnect = chain
-
     # Make sure it's valid
-    if chain.hasExpired()['Value']:
+    if chain.hasExpired().get('Value'):
       return S_ERROR("Proxy %s has expired" % proxyLocation)
+    if chain.getDIRACGroup().get('Value') or chain.isVOMS().get('Value'):
+      return S_ERROR("Cannot upload proxy with DIRAC group or VOMS extensions")
 
-    #rpcClient = RPCClient( "Framework/ProxyManager", proxyChain = chainToConnect )
     rpcClient = RPCClient("Framework/ProxyManager", timeout=120)
     # Get a delegation request
-    result = rpcClient.requestDelegationUpload(chain.getRemainingSecs()['Value'], diracGroup)
+    result = rpcClient.requestDelegationUpload(chain.getRemainingSecs()['Value'])#, diracGroup)
     if not result['OK']:
       return result
-    # Check if the delegation has been granted
-    if 'Value' not in result or not result['Value']:
-      if 'proxies' in result:
-        return S_OK(result['proxies'])
-      else:
-        return S_OK()
     reqDict = result['Value']
     # Generate delegated chain
     chainLifeTime = chain.getRemainingSecs()['Value'] - 60
     if restrictLifeTime and restrictLifeTime < chainLifeTime:
       chainLifeTime = restrictLifeTime
     retVal = chain.generateChainFromRequestString(reqDict['request'],
                                                   lifetime=chainLifeTime,
-                                                  diracGroup=diracGroup, rfc=rfcIfPossible)
+                                                  rfc=rfcIfPossible)
     if not retVal['OK']:
       return retVal
     # Upload!
     result = rpcClient.completeDelegationUpload(reqDict['id'], retVal['Value'])
     if not result['OK']:
       return result
-    if 'proxies' in result:
-      return S_OK(result['proxies'])
-    return S_OK()
+    return S_OK(result.get('proxies') or result['Value'])
 
+  # FIXME: Old method
```

Comment:
If there are old methods that are unused within DIRAC (BTW did you check WebAppDIRAC?) then don't put a #FIXME, rather use the `deprecated` decorator (search the code for it)

Reply:
It's my work comment that I forgot to remove.

Reply:
Yes, I remember about WebAppDIRAC. I`ll do it.